### PR TITLE
Fixed option check when proposing wpa supplicant

### PR DESCRIPTION
**Pull request by mchf**
Bnc1089471

Fixes https://github.com/yast/yast-network/pull/622

### CODE REVIEW

File: test/lan_test.rb [hunk 1]
```diff
@@ -7,6 +7,26 @@
 Yast.import "Lan"
 
 describe "LanClass#Packages" do
+  it "do not list wpa_supplicant package when WIRELESS_AUTH_MODE is not psk or eap" do
```
Comment (teclator) [inline test/lan_test.rb]: does

File: package/yast2-network.changes [hunk 2]
```diff
@@ -1,3 +1,10 @@
+-------------------------------------------------------------------
+Mon Apr 23 18:10:29 UTC 2018 - mfilka@suse.com
+
+- bnc#1089471 
+  - fixed option check to get proper proposal
```
Comment (teclator) [inline package/yast2-network.changes]: The changelog could be more descriptive about the problem and solution, read it alone without context...

File: test/lan_test.rb [hunk 3]
```diff
@@ -7,6 +7,26 @@
 Yast.import "Lan"
 
 describe "LanClass#Packages" do
+  it "does not list wpa_supplicant package when WIRELESS_AUTH_MODE is not psk or eap" do
+    allow(Yast::NetworkInterfaces)
```
Comment (teclator) [inline test/lan_test.rb]: This mock seems to be useless

File: test/lan_test.rb [hunk 4]
```diff
@@ -7,6 +7,26 @@
 Yast.import "Lan"
 
 describe "LanClass#Packages" do
+  it "does not list wpa_supplicant package when WIRELESS_AUTH_MODE is not psk or eap" do
```
Comment (teclator) [inline test/lan_test.rb]: time to clean up it a little and reuse some mocks? what about start adding some context and use let to define some of the attributes?

File: test/lan_test.rb [hunk 5]
```diff
@@ -7,46 +7,60 @@
...
-      .with("wpa_supplicant")
-      .at_least(:once)
-      .and_return(false)
-    expect(Yast::Lan.Packages).to include "wpa_supplicant"
+  context "When NetworkManager is not going to be installed" do
+    before(:each) do
+      allow(Yast::NetworkService)
+        .to receive(:is_network_manager)
+        .and_return(false)
+      expect(Yast::PackageSystem)
```
Comment (teclator) [inline test/lan_test.rb]: It should be an allow

File: test/lan_test.rb [hunk 6]
```diff
@@ -7,46 +7,60 @@
...
-      .and_return(true)
-    expect(Yast::PackageSystem)
-      .to receive(:Installed)
-      .with("NetworkManager")
-      .at_least(:once)
-      .and_return(false)
-    expect(Yast::Lan.Packages).to include "NetworkManager"
+  context "When NetworkManager is going to be installed" do
+    it "lists NetworkManager package" do
+      allow(Yast::NetworkInterfaces)
```
Comment (teclator) [inline test/lan_test.rb]: Not needed

File: test/lan_test.rb [hunk 7]
```diff
@@ -7,46 +7,60 @@
...
+        .to receive(:List)
+        .and_return([])
+      allow(Yast::NetworkInterfaces)
```
Comment (teclator) [inline test/lan_test.rb]: Not needed

File: test/lan_test.rb [hunk 8]
```diff
@@ -7,46 +7,53 @@
...
-      .with("wpa_supplicant")
-      .at_least(:once)
-      .and_return(false)
-    expect(Yast::Lan.Packages).to include "wpa_supplicant"
+  context "When NetworkManager is not going to be installed" do
+    before(:each) do
+      allow(Yast::NetworkService)
+        .to receive(:is_network_manager)
+        .and_return(false)
+      allow(Yast::PackageSystem)
```
[the remark this answers is not in the document]
Comment (mchf) [reply test/lan_test.rb]: such test is currently missing and I'd stop with adding more improvements as the issue currently block openqa

**Pull request merged**